### PR TITLE
easytier: init at 1.2.3

### PR DESCRIPTION
**Pull request by L-Trump**
## Description of changes

EasyTier is an open-source, simple, safe and decentralized VPN networking solution implemented with the Rust language and Tokio framework.
Homepage: https://github.com/EasyTier/EasyTier

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by L-Trump** [comment]
```bash
> nix build .#easytier
> tree result
result
└── bin
    ├── easytier-cli
    └── easytier-core
```

File: pkgs/by-name/ea/easytier/package.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  cargoHash = "sha256-9wAGUVYKz7+Q8y+dmay8pEZnv7PikzuAoas/h5T3sLE=";
+
+  nativeBuildInputs = [ protobuf ];
+
+  buildNoDefaultFeatures = stdenv.isMips;
+  buildFeatures = lib.optional stdenv.isMips [ "mips" ] ++ lib.optional withQuic [ "quic" ];
+
+  doCheck = false; # tests failed due to heavy rely on network
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/ea/easytier/package.nix:32]:
```suggestion
  meta = {
```

File: pkgs/by-name/ea/easytier/package.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    homepage = "https://github.com/EasyTier/EasyTier";
+    description = "Simple, decentralized mesh VPN with WireGuard support.";
```
Comment (Aleksanaa) [inline pkgs/by-name/ea/easytier/package.nix:34]:
```suggestion
    description = "Simple, decentralized mesh VPN with WireGuard support";
```

File: pkgs/by-name/ea/easytier/package.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    longDescription = ''
+      EasyTier is a simple, safe and decentralized VPN networking solution implemented
+      with the Rust language and Tokio framework.
+    '';
+    license = licenses.asl20;
+    platforms = with platforms; unix ++ windows;
+    maintainers = with maintainers; [ ltrump ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ea/easytier/package.nix:41]: missing `mainProgram`

File: pkgs/by-name/ea/easytier/package.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  };
+
+  cargoHash = "sha256-9wAGUVYKz7+Q8y+dmay8pEZnv7PikzuAoas/h5T3sLE=";
+
+  nativeBuildInputs = [ protobuf ];
+
+  buildNoDefaultFeatures = stdenv.isMips;
+  buildFeatures = lib.optional stdenv.isMips [ "mips" ] ++ lib.optional withQuic [ "quic" ];
+
+  doCheck = false; # tests failed due to heavy rely on network
```
Comment (Aleksanaa) [inline pkgs/by-name/ea/easytier/package.nix:30]: Can we only disable some of them? See https://nixos.org/manual/nixpkgs/unstable/#disabling-package-tests

File: pkgs/by-name/ea/easytier/package.nix (hunk 1 above)
Comment (L-Trump) [reply pkgs/by-name/ea/easytier/package.nix]: Done

File: pkgs/by-name/ea/easytier/package.nix (hunk 4 above)
Comment (L-Trump) [reply pkgs/by-name/ea/easytier/package.nix:30]: Thx for your review! Almost all of the tests for this program rely heavily on the network or on the target host's device. E.g., if the target host does not support IPV6, all related tests will fail. I was able to pass the offline check on my host by skipping nearly half of the tests. but I'm not sure if it works on other hosts.

**Comment by L-Trump** [comment]
@yunfachi Done. Thanks for review!

**Comment by yunfachi** [comment]
Almost done. The commit `maintainers: add ltrump` currently comes after `easytier: init at 1.2.3`, but it should be before it.

**Comment by L-Trump** [comment]
@yunfachi Done. Sorry for my misunderstanding

**Review by Aleksanaa — APPROVED** [review]

**Review by yunfachi — APPROVED** [review]

**Comment by L-Trump** [comment]
Is there anything I need to do next?

**Comment by L-Trump** [comment]
@Bot-wxt1221 Done. Thanks for your review!

**Review by Aleksanaa — APPROVED** [review]

**Comment by gshpychka** [comment]
Result of `nixpkgs-review pr 339616` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>easytier</li>
  </ul>
</details>

**Comment by gshpychka** [comment]
> Result of `nixpkgs-review pr 339616` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 
> 1 package failed to build:

```
error: builder for '/nix/store/9qay4s6gi0n6rkhrgs42hpk06szr6w5g-easytier-1.2.3.drv' failed with exit code 101;
       last 10 log lines:
       > warning: build failed, waiting for other jobs to finish...
       > error: linking with `/nix/store/8c3v3rdzvi5364x9m959ijyy6iamqf5f-clang-wrapper-16.0.6/bin/cc` failed: exit status: 1
       >   |
       >   = note: env -u IPHONEOS_DEPLOYMENT_TARGET -u TVOS_DEPLOYMENT_TARGET -u XROS_DEPLOYMENT_TARGET LC_ALL="C" PATH="/nix/store/936pxmssyhsdamkagi1ir7cxz0ldfkql-rustc-1.80.1/lib/rustlib/aarch64-apple-darwin/bin:/nix/store/936pxmssyhsdamkagi1ir7cxz0ldfkql-rustc-1.80.1/lib/rustlib/aarch64-apple-darwin/bin:/nix/store/936pxmssyhsdamkagi1ir7cxz0ldfkql-rustc-1.80.1/lib/rustlib/aarch64-apple-darwin/bin:/nix/store/idfq37bavykpfahg0q8cpawj90sckagq-cargo-1.80.1/bin:/nix/store/wnllanmkgvky57cd0p082w4y99x7m8zd-cargo-auditable-0.6.2/bin:/nix/store/bh26nkj0y3q5a8mxb23w3ajbsa113kqq-protobuf-25.4/bin:/nix/store/s0spffswbklzf4jj4hg2b0d9wxyig67j-auditable-cargo-1.80.1/bin:/nix/store/idfq37bavykpfahg0q8cpawj90sckagq-cargo-1.80.1/bin:/nix/store/3gggf0knmsc6srxq89gm424rffwma6rv-rustc-wrapper-1.80.1/bin:/nix/store/8c3v3rdzvi5364x9m959ijyy6iamqf5f-clang-wrapper-16.0.6/bin:/nix/store/ydjx24wmcpcgkjgn0l1nr1hgb2adny7b-clang-16.0.6/bin:/nix/store/1l41471x6rlf7l544s1bxkqqgpprd28m-coreutils-9.5/bin:/nix/store/gd1a3hsa7mdvlijbx6d1bibjfn215nbk-cctools-binutils-darwin-wrapper-1010.6/bin:/nix/store/8qxrkah3ylxzqbpgilmyqvrddrzzyr63-cctools-binutils-darwin-1010.6/bin:/nix/store/1l41471x6rlf7l544s1bxkqqgpprd28m-coreutils-9.5/bin:/nix/store/6iir6ljd4yan81glr5wh3zc9x5l0fwdj-findutils-4.10.0/bin:/nix/store/44byadd2lwqid70aarkjdxqm4bnamiff-diffutils-3.10/bin:/nix/store/671pvppgcdzgdxkbhggb5z1zs3dx096y-gnused-4.9/bin:/nix/store/p57w57nln7yjvj95i9wlx27g7gcnwgga-gnugrep-3.11/bin:/nix/store/cqaf3cx80idg9xrqca65rpm3ax8m2xpq-gawk-5.2.2/bin:/nix/store/8pb64yz6kvipynxwbmj8v3mg42r2ainb-gnutar-1.35/bin:/nix/store/11q17b4qydb1z4lksslmz592ld1fphbb-gzip-1.13/bin:/nix/store/xjfp4zg4vz80yc95zfczw5vc09jjcrgn-bzip2-1.0.8-bin/bin:/nix/store/6mcdimq65bpmk4ccsbwk8zqqv0w57xki-gnumake-4.4.1/bin:/nix/store/iafzjk7zbkqaszqfp6n006vvxjrpn4f6-bash-5.2p32/bin:/nix/store/88lm1zpf550i9jl14nkhmrwfpk57h53h-patch-2.7.6/bin:/nix/store/7vh6m6xky6har0cydkpmlq81l0yqzlbq-xz-5.6.2-bin/bin:/nix/store/ciivdvv31qrccs59v9hryac5jzm53qwi-file-5.45/bin" VSLANG="1033" ZERO_AR_DATE="1" "/nix/store/8c3v3rdzvi5364x9m959ijyy6iamqf5f-clang-wrapper-16.0.6/bin/cc" "-arch" "arm64" "/private/tmp/nix-build-easytier-1.2.3.drv-0/rustcG787jd/symbols.o" "/private/tmp/nix-build-easytier-1.2.3.drv-0/source/target/aarch64-apple-darwin/release/deps/easytier_core-8cb86b410d44c671.easytier_core.3b6b38a28a3c5e83-cgu.0.rcgu.o" "-L" "/private/tmp/nix-build-easytier-1.2.3.drv-0/source/target/aarch64-apple-darwin/release/deps" "-L" "/private/tmp/nix-build-easytier-1.2.3.drv-0/source/target/release/deps" "-L" "/private/tmp/nix-build-easytier-1.2.3.drv-0/source/target/aarch64-apple-darwin/release/build/ring-01761401bd6a7ecd/out" "-L" "/private/tmp/nix-build-easytier-1.2.3.drv-0/source/target/aarch64-apple-darwin/release/build/mimalloc-rust-sys-e949de6902c24768/out" "-L" "/private/tmp/nix-build-easytier-1.2.3.drv-0/source/target/aarch64-apple-darwin/release/build/network-interface-74adab19f498a986/out" "-L" "/private/tmp/nix-build-easytier-1.2.3.drv-0/source/target/aarch64-apple-darwin/release/build/ring-537c4a3c997aaf79/out" "-L" "/nix/store/936pxmssyhsdamkagi1ir7cxz0ldfkql-rustc-1.80.1/lib/rustlib/aarch64-apple-darwin/lib" "/private/tmp/nix-build-easytier-1.2.3.drv-0/rustcG787jd/libring-ddd0cad00f9eb221.rlib" "/private/tmp/nix-build-easytier-1.2.3.drv-0/rustcG787jd/libmimalloc_rust_sys-e97e30b45e15cb0a.rlib" "/private/tmp/nix-build-easytier-1.2.3.drv-0/rustcG787jd/libnetwork_interface-7b0a2f20fd11cedf.rlib" "/private/tmp/nix-build-easytier-1.2.3.drv-0/rustcG787jd/libring-3c23d3bf5abdfb5c.rlib" "/nix/store/936pxmssyhsdamkagi1ir7cxz0ldfkql-rustc-1.80.1/lib/rustlib/aarch64-apple-darwin/lib/libcompiler_builtins-49429d6710111059.rlib" "-framework" "CoreFoundation" "-framework" "Security" "-framework" "CoreFoundation" "-liconv" "-lSystem" "-lc" "-lm" "-L" "/nix/store/936pxmssyhsdamkagi1ir7cxz0ldfkql-rustc-1.80.1/lib/rustlib/aarch64-apple-darwin/lib" "-o" "/private/tmp/nix-build-easytier-1.2.3.drv-0/source/target/aarch64-apple-darwin/release/deps/easytier_core-8cb86b410d44c671" "-Wl,-dead_strip" "-nodefaultlibs" "/private/tmp/nix-build-easytier-1.2.3.drv-0/source/target/aarch64-apple-darwin/release/deps/easytier_core_audit_data.o" "-Wl,-u,_AUDITABLE_VERSION_INFO"
       >   = note: ld: framework not found Security
       >           clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
       >
       >
       > warning: `easytier` (bin "easytier-core") generated 1 warning (1 duplicate)
       > error: could not compile `easytier` (bin "easytier-core") due to 1 previous error; 1 warning emitted
       For full logs, run 'nix log /nix/store/9qay4s6gi0n6rkhrgs42hpk06szr6w5g-easytier-1.2.3.drv'.
error: 1 dependencies of derivation '/nix/store/d8gkpb23gk7ib52rrynma3y1hm0nyj1j-review-shell.drv' failed to build
```

File: pkgs/by-name/ea/easytier/package.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  rustPlatform,
+  protobuf,
+  nix-update-script,
```
Comment (afh) [inline pkgs/by-name/ea/easytier/package.nix:7]:
```suggestion
  nix-update-script,
  darwin,
```

File: pkgs/by-name/ea/easytier/package.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  src = fetchFromGitHub {
+    owner = "EasyTier";
+    repo = "EasyTier";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-7T6xdJrVTgg7rSTC2PaVTsBTgi14qJzaR6M8tRUL8OQ=";
+  };
+
+  cargoHash = "sha256-9wAGUVYKz7+Q8y+dmay8pEZnv7PikzuAoas/h5T3sLE=";
+
+  nativeBuildInputs = [ protobuf ];
```
Comment (afh) [inline pkgs/by-name/ea/easytier/package.nix:24]:
```suggestion
  nativeBuildInputs = [ protobuf ];
  
  buildInputs = lib.optionals stdenv.isDarwin [
    darwin.apple_sdk.frameworks.Security
  ];
```

**Review by afh — CHANGES_REQUESTED** [review]
Please find below a few suggestions that will address the darwin build failure reported previously.

**Comment by L-Trump** [comment]
@gshpychka Done. Thanks to the suggestions from @afh !!!

**Comment by Aleksanaa** [comment]
> Please find below a few suggestions that will address the darwin build failure reported previously.

Have you confirmed that only this one is missing?

**Comment by uncenter** [comment]
Result of `nixpkgs-review pr 339616` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>easytier</li>
  </ul>
</details>

**Comment by viraptor** [comment]
Result of `nixpkgs-review pr 339616` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>easytier</li>
  </ul>
</details>

Seems to start fine.

**Review by Aleksanaa — APPROVED** [review]

**Comment by afh** [comment]
@Aleksanaa, yes I have that the Security framework is the needed framework dependency by building it locally 🙂

**Pull request merged**